### PR TITLE
Render shows to homepage

### PR DESCRIPTION
**Pull request by ahmedalbaiti**
Created shows controller with index action and the accompanying view, and created the spec to test it

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,6 +3,8 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby "3.1.2"
 
+gem "pry-byebug"
```
Comment (haydon-rh) [inline Gemfile]: you might move `pry` to development and testing!

File: app/views/shows/index.html.haml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+- @shows.each do |show|
+	%h1= show["title"]
+	%h3= show["type"]
+	%end 
+
```
Comment (haydon-rh) [inline app/views/shows/index.html.haml]: Your newline might not have worked here if it has indentation on it. Github still thinks there's no empty line at the end of the file

File: spec/root_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+RSpec.describe 'index route', type: :system do
+  it 'shows page content' do
+    visit '/'
+    expect(page).to have_content("Dick Johnson Is Dead") 
+    expect(page).to have_content("Blood & Water") 
+    expect(page).to have_content("Ganglands") 
+    expect(page).to have_content("Jailbirds New Orleans") 
+    expect(page).to have_content("Midnight Mass")
+    expect(page).to have_content("Sankofa")
```
Comment (haydon-rh) [inline spec/root_spec.rb]: These tests seem to be testing state instead of logic. It might not be a bad thing at this point in time, but the test would also pass if in your controller, you rendered all first 10 titles in your CSV instead of reading from the CSV itself. As a more robust test, you could potentially stub `CSV` here, and have it return an array of 11 items, and then test that only the first 10 are displayed.

Because you haven't built a file parser here, testing this is a little bit more difficult than the way we do it in the codebase, but this test will definitely need to change as you start doing more complex things with your show data.

File: app/controllers/shows_controller.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'csv'
+require 'pry-byebug'
+class ShowsController < ApplicationController
+    def index 
+    	@shows = CSV.parse(File.read('lib/netflix_titles.csv'), headers: true).first(10)
```
Comment (haydon-rh) [inline app/controllers/shows_controller.rb]: there are a couple of magic variables here - perhaps `10` could move to a constant called `NUMBER_OF_SHOWS` and your file path could move to a variable called `show_titles_file`?

it would then look like this:
```
@shows = CSV.parse(File.read(show_titles_file), headers:true).first(NUMBER_OF_SHOWS)
```
WDYT?

**Review by haydon-rh — COMMENTED** [review]
Good start! A couple of changes suggested, but up to you on your implementation going forward. Nice 👍

File: spec/root_spec.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'index route', type: :system do
```
Comment (smart-jones) [inline spec/root_spec.rb]: This looks like a cross between controller/view/request/feature spec. Familiarise yourself with the following guidelines to create tests that closely aligns with these examples. 

https://relishapp.com/rspec/rspec-rails/docs/controller-specs
https://relishapp.com/rspec/rspec-rails/v/6-0/docs/view-specs/view-spec
https://relishapp.com/rspec/rspec-rails/v/6-0/docs/feature-specs/feature-spec
https://relishapp.com/rspec/rspec-rails/v/6-0/docs/request-specs/request-spec

File: spec/requests/shows_spec.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe "Shows", type: :request do
+  describe "GET /index" do
+    pending "add some examples (or delete) #{__FILE__}"
```
Comment (smart-jones) [inline spec/requests/shows_spec.rb]: You can delete these pending tests, if you don't need them

File: spec/helpers/shows_helper_spec.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+#
+# describe ShowsHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe ShowsHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (smart-jones) [inline spec/helpers/shows_helper_spec.rb]: Same here, you can delete these files if you don't them at present.

File: app/views/shows/index.html.haml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+- @shows.each do |show|
```
Comment (smart-jones) [inline app/views/shows/index.html.haml:1]: Indentation is screwed up here

File: app/helpers/shows_helper.rb [hunk 9]
```diff
@@ -0,0 +1,2 @@
+module ShowsHelper
```
Comment (smart-jones) [inline app/helpers/shows_helper.rb]: You can remove these empty files

File: app/controllers/shows_controller.rb [hunk 10]
```diff
@@ -0,0 +1,7 @@
+require 'csv'
+require 'pry-byebug'
+class ShowsController < ApplicationController
+    def index
```
Comment (smart-jones) [inline app/controllers/shows_controller.rb]: Indentation is screwed up here. Make sure you install and run Rubocop (and rucocop-rspec) over your entire project

File: Gemfile.lock [hunk 11]
```diff
@@ -91,6 +93,10 @@ GEM
     erubi (1.11.0)
     globalid (1.0.0)
       activesupport (>= 5.0)
+    haml (6.0.10)
```
Comment (smart-jones) [inline Gemfile.lock:97]: Haml is not the best language for learning. It can get confusing quite quickly. ERB will do just fine, but it's just an opinion

**Review by smart-jones — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 1 above)
Comment (ahmedalbaiti) [reply Gemfile]: noted

File: Gemfile.lock (hunk 11 above)
Comment (ahmedalbaiti) [reply Gemfile.lock:97]: Yeah, I agree, but I'm simply following the requirements on Techmap for this

File: app/controllers/shows_controller.rb (hunk 10 above)
Comment (ahmedalbaiti) [reply app/controllers/shows_controller.rb]: Done

File: app/controllers/shows_controller.rb (hunk 4 above)
Comment (ahmedalbaiti) [reply app/controllers/shows_controller.rb]: Agree and done

File: spec/root_spec.rb (hunk 5 above)
Comment (ahmedalbaiti) [reply spec/root_spec.rb]: Changed the test, does it look better now?

File: spec/root_spec.rb [hunk 12]
```diff
@@ -7,6 +7,6 @@
   it 'shows list of first 10 shows' do
     assign :shows, @shows = CSV.parse(File.read('lib/netflix_titles.csv'), headers: true).first(10)
     render
-    @shows.each { |show| expect(rendered).to have_content(show["title"]) }   
+    @shows.each { |show| expect(rendered).to have_content(show['title']) }
```
Comment (smart-jones) [inline spec/root_spec.rb]: `spec/root_spec.rb` you need a category i.e. `spec/controller/shows_spec.rb`

Comment (ahmedalbaiti) [reply spec/root_spec.rb]: Done

File: spec/root_spec.rb (hunk 3 above)
Comment (ahmedalbaiti) [reply spec/root_spec.rb]: > These tests seem to be testing state instead of logic. It might not be a bad thing at this point in time, but the test would also pass if in your controller, you rendered all first 10 titles in your CSV instead of reading from the CSV itself. As a more robust test, you could potentially stub `CSV` here, and have it return an array of 11 items, and then test that only the first 10 are displayed.
> 
> Because you haven't built a file parser here, testing this is a little bit more difficult than the way we do it in the codebase, but this test will definitely need to change as you start doing more complex things with your show data.

So basically in my latest build I haven't stubbed the file but I tested the logic of reading from the CSV itself

File: app/controllers/shows_controller.rb [hunk 13]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'pry-byebug'
+
+NUMBER_OF_SHOWS = 10
+class ShowsController < ApplicationController
+  def index
+    @shows = CSV.parse(File.read('lib/netflix_titles.csv'), headers: true).first(NUMBER_OF_SHOWS)
```
Comment (Swiftda01) [inline app/controllers/shows_controller.rb:9]:
```suggestion
    @shows = CSV.read('lib/netflix_titles.csv', headers: true).first(NUMBER_OF_SHOWS)
```
Optional: Avoid using `File` class needlessly

Comment (ahmedalbaiti) [reply app/controllers/shows_controller.rb:9]: Done, and also added a fixture

**Review by Swiftda01 — APPROVED** [review]
